client/dashboard/sites/overview/site-card.tsx

```diff
@@ -102,40 +103,70 @@ function FieldTitle( { children }: { children: React.ReactNode } ) {
 	);
 }
 
-function PlanDetails( { site, currentPlan }: { site: Site; currentPlan: Plan } ) {
-	if ( ! site.plan || ! currentPlan ) {
+function PlanDetails( { site }: { site: Site } ) {
+	const { data: currentPlan } = useQuery( siteCurrentPlanQuery( site.slug ) );
+
+	if ( ! site.plan ) {
 		return null;
 	}
 
 	const {
 		plan: { product_name_short, is_free: isFree },
 	} = site;
-	const { expiry, id } = currentPlan;
+
 	return (
 		<VStack>
 			<FieldTitle>{ __( 'Plan' ) }</FieldTitle>
 			{ product_name_short && <Text>{ product_name_short }</Text> }
-			<Text>{ getPlanExpirationMessage( { isFree, expiry } ) }</Text>
-			{ id ? (
-				<Button href={ `/purchases/subscriptions/${ site.slug }/${ id }` } variant="link">
-					{ __( 'Manage subscription' ) }
-				</Button>
-			) : (
+			<Text>{ getPlanExpirationMessage( { isFree, currentPlan } ) }</Text>
+			{ isFree ? (
 				<Button href={ `/plans/${ site.slug }` } variant="link">
 					{ __( 'Upgrade' ) }
 				</Button>
+			) : (
+				<Button
+					// @ts-expect-error inert is not typed
+					inert={ ! currentPlan }
+					href={ currentPlan ? `/purchases/subscriptions/${ site.slug }/${ currentPlan.id }` : '' }
+					variant="link"
+				>
+					{ currentPlan ? (
+						__( 'Manage subscription' )
+					) : (
+						// Show a blurred text while we wait for the plan ID.
+						<TextBlur text={ __( 'Manage subscription' ) } />
+					) }
+				</Button>
 			) }
 		</VStack>
 	);
 }
 
-function getPlanExpirationMessage( { isFree, expiry }: { isFree: boolean; expiry?: string } ) {
+function getPlanExpirationMessage( {
+	isFree,
+	currentPlan,
+}: {
+	isFree: boolean;
+	currentPlan?: Plan;
+} ) {
 	if ( isFree ) {
 		return __( 'No expiration date.' );
 	}
-	return (
-		expiry &&
-		/* translators: %s: date of plan's expiration date. Eg.  August 20, 2025 */
-		sprintf( __( 'Expires on %s.' ), dateI18n( 'F j, Y', expiry ) )
+
+	/* translators: %s: date of plan's expiration date. Eg.  August 20, 2025 */
+	const expiresString = __( 'Expires on <time>%s</time>.' );
+
+	// Show a blurred time while we wait for the current plan.
```

Comment:
This assumes that every plan has expiry, which is not true. For example, sites created via A4A won't have that, which breaks in this PR:

<img width="331" alt="image" src="https://github.com/user-attachments/assets/7ead4175-5d3d-4596-b0d7-c6f928e2f0b8" />

Reply:
It's `null` in such cases.

I believe it's already reflected in the type:

https://github.com/Automattic/wp-calypso/blob/05935dbf90195ceebbf10f8712ed3274c37b49fd/client/dashboard/data/types.ts#L58

Reply:
That would be `undefined`, no?

Reply:
I fixed it in [ad4fcee](https://github.com/Automattic/wp-calypso/pull/103601/commits/ad4fceef0c9ade3fbda341d908bf8c4e60df0b8c)

Reply:
But it means there will still be layout shift for such sites.

Maybe that's OK for now; in general I believe A4A sites should be handled / shown differently in the overview page.

Reply:
> That would be undefined, no?

I always forgot the difference between `?` and `null` 🤦

Reply:
@fushar Ah you're right. That's unacceptable to me... 😄 so to me that's a blocker for this PR. Ideally we'd know beforehand when there would be no expiration date, like with the free plan. Not sure if that's possible. Alternatively we should show "No expiration date" here, instead of null, but that would still mean a little shift from "Expires on ..." to "No expiration date", which is not ideal.

Reply:
> Alternatively we should show "No expiration date" here

I think it's semantically wrong, because the expiration / subscription actually exists but is managed by A4A 🥲 

Maybe we can blur a whole line ( random sentence), then just make it disappear for such cases 😄

Reply:
OR we blur "Expires on ..." entirely so it can be whatever it needs to be.

Reply:
Ok, I blurred the entire line instead.

Reply:
> I think it's semantically wrong, because the expiration / subscription actually exists but is managed by A4A 🥲

Wait, so doesn't the "manage subscription" link then also not make sense? I'm curious what info there is in the `site.plan` when it's an A4A site.

Reply:
Ah right... seems so. BTW, this is how the site overview looks for A4A site in v1:

<img width="672" alt="image" src="https://github.com/user-attachments/assets/0c0c944d-bb38-4272-ab04-d7f76c2a8f31" />

Perhaps we shouldn't worry about it now...

Reply:
How do you check if it's managed? Is that in the plan? I don't have a site that is a4a atm, not sure how to test

Reply:
Something like this 🙈 

https://github.com/Automattic/wp-calypso/blob/474d76d5fe3f1d12e17bdd26a911405bde51f90c/client/lib/purchases/index.ts#L890-L892